### DISCUSSION TITLE
"Deny unknown" option for `Query` extractor?

### DISCUSSION THREAD

**Discussion by cyqsimon** [actix/actix-web#3466] [Q&A] [answered]
Is this something that actix supports? I would like my API to fail fast when a client attempts to use it with incorrect query parameters.

This is particularly useful when the API introduces breaking changes in the form of a query parameter rename. Instead of silently ignoring the old parameter and reverting back to some default behaviour (which is an accident waiting to happen), it may be best, depending on the exact circumstance, for the API to immediately error out.

**Comment by robjtede** [comment, marked as answer]
This is supported via standard serde attributes. See: https://serde.rs/container-attrs.html#deny_unknown_fields

**Reply by cyqsimon** [reply]
Ah yes big thanks. It makes complete sense once you mention it.
